Realm/ObjectServerTests/RLMAsymmetricSyncServerTests.mm

```diff
@@ -0,0 +1,163 @@
+    configuration.objectClasses = @[PersonAsymmetric.self];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
+    [realm beginWriteTransaction];
+    PersonAsymmetric *person = [[PersonAsymmetric alloc] initWithPrimaryKey:[RLMObjectId objectId]
+                                                                  firstName:[NSString stringWithFormat:@"firstname"]
+                                                                   lastName:[NSString stringWithFormat:@"lastname"]
+                                                                        age:20];
+    [realm createObject:person];
+    [realm commitWriteTransaction];
+    [self checkCountInMongo:0];
+}
+
+- (void)testCreateAsymmetricObject {
+    RLMUser *user = [self flexibleSyncUser:_cmd];
+    RLMRealmConfiguration *configuration = [user flexibleSyncConfiguration];
+    configuration.objectClasses = @[PersonAsymmetric.self];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
+    XCTAssertNotNil(realm);
+
+    [realm beginWriteTransaction];
+    const int numberOfSubs = 12;
+    for (int i = 1; i <= numberOfSubs; ++i) {
+        PersonAsymmetric *person = [[PersonAsymmetric alloc] initWithPrimaryKey:[RLMObjectId objectId]
+                                                                      firstName:[NSString stringWithFormat:@"firstname_%d", i]
+                                                                       lastName:[NSString stringWithFormat:@"lastname_%d", i]
+                                                                            age:i];
+        [realm createObject:person];
+    }
+    [realm commitWriteTransaction];
+    [self waitForUploadsForRealm:realm];
+    [self checkCountInMongo:12];
+}
+
+- (void)testCreateAsymmetricSameObjectNotDuplicates {
+    RLMUser *user = [self flexibleSyncUser:_cmd];
+    RLMRealmConfiguration *configuration = [user flexibleSyncConfiguration];
+    configuration.objectClasses = @[PersonAsymmetric.self];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
+    XCTAssertNotNil(realm);
+
+    PersonAsymmetric *person = [[PersonAsymmetric alloc] initWithPrimaryKey:[RLMObjectId objectId]
+                                                                  firstName:[NSString stringWithFormat:@"firstname"]
+                                                                   lastName:[NSString stringWithFormat:@"lastname"]
+                                                                        age:10];
+    [realm beginWriteTransaction];
+    [realm createObject:person];
+    [realm commitWriteTransaction];
+    [self waitForUploadsForRealm:realm];
+    [self checkCountInMongo:1];
+
+    [realm beginWriteTransaction];
+    [realm createObject:person];
+    [realm commitWriteTransaction];
+    [self waitForUploadsForRealm:realm];
+    [self checkCountInMongo:1];
+}
+@end
```

Comment:
We should also test that Asymmetric objects with invalid schemas throw.

Reply:
Added a few tests for links within an asymmetric object and a top level object linking an asymmetric object.   The validation is done by core and the errors are thrown when the realm is opened.